rscripts/YAL5-codon-pairs.R

```diff
@@ -0,0 +1,1455 @@
+                                                       hd_file,
+                                                       min_read_length,
+                                                       snapdisp,
+                                                       asite_disp_path,
+                                                       filter_for_frame)
+
+### FilterForFeaturePositions DOES NOT APPREAR TO BE USED ANYWHERE
+
+# interesting_feature_positions <- FilterForFeaturePositions(codon_pos_pair_i200,
+#                                                            gene_names,
+#                                                            dataset,
+#                                                            hd_file,
+#                                                            min_read_length,
+#                                                            snapdisp,
+#                                                            asite_disp_path,
+#                                                            filter_for_frame,
+#                                                            feature_of_interest)
+
+
+expand_feature_region <- ExpandFeatureRegion(codon_pos_pair_i200,
+                                             gene_names,
+                                             dataset,
+                                             hd_file,
+                                             gff_df,
+                                             min_read_length,
+                                             snapdisp,
+                                             asite_disp_path,
+                                             filter_for_frame,
+                                             feature_of_interest,
+                                             expand_width)
+
+normalized_expand_list <- ExpandedRegionNormalisation(codon_pos_pair_i200,
+                                                      gene_names, dataset, hd_file,
+                                                      gff_df, min_read_length, snapdisp,
+                                                      asite_disp_path,
+                                                      filter_for_frame,
+                                                      feature_of_interest,
+                                                      expand_width)
+
+overlayed_tibbles <- OverlayedTibble(codon_pos_pair_i200,
+                                     gene_names, dataset, hd_file,
+                                     gff_df, min_read_length,
+                                     snapdisp, asite_disp_path,
+                                     filter_for_frame, feature_of_interest,
+                                     expand_width)
+
+plot <- GeneratePlot(codon_pos_pair_i200, gene_names, dataset, hd_file,
+                     gff_df, min_read_length, snapdisp, asite_disp_path,
+                     filter_for_frame, feature_of_interest,
+                     expand_width, size = 12)
```

Comment:
Because all of the functions are nested, the whole script could be run with just the `GeneratePlot` function. When running larger datasets it would be more efficient to have the other commands as comments, or remove them, so the same steps aren't repeated multiple times